File: src/main/java/net/nerfatg/proxy/packet/packets/ReadyPlayerCount.java
package net.nerfatg.proxy.packet.packets;

import net.nerfatg.proxy.packet.Packet;
import net.nerfatg.proxy.packet.PacketAction;
import net.nerfatg.proxy.packet.PacketType;

import java.nio.BufferOverflowException;
import java.nio.ByteBuffer;

public class ReadyPlayerCount extends Packet {
    private byte readyPlayers;

    public ReadyPlayerCount(ByteBuffer buffer) throws BufferOverflowException {
        super(buffer);
    }

    public ReadyPlayerCount(byte readyPlayers, PacketAction action) {
        super(PacketType.ReadyPlayerCount, action);
        this.readyPlayers = readyPlayers;
    }

    @Override
    public void readPayload(ByteBuffer buffer, int offset) throws BufferOverflowException {
        buffer.position(offset);
        this.readyPlayers = buffer.get();
    }

    @Override
    public void writePayload(ByteBuffer buffer, int offset) throws BufferOverflowException {
        buffer.position(offset);
        buffer.put(readyPlayers);
    }

    public byte getReadyPlayers() { return readyPlayers; }

    @Override
    public String toString() {
        return String.format("ReadyPlayerCount{ readyPlayerCount=%d }", readyPlayers);
    }
}

File: Packets/packets/ReadyPlayerCount.cs
using System;
using System.Text;


public class ReadyPlayerCount : Packet<PacketType>
{
    public byte ReadyPlayers { get; private set; }

    public ReadyPlayerCount(byte[] bytes) : base(bytes) { }

    public ReadyPlayerCount(byte readyPlayerCount, PacketAction action) : base(PacketType.ReadyPlayerCount, action)
    {
        this.ReadyPlayers = readyPlayerCount;
    }

    protected override void ReadPayload(byte[] bytes, byte offset)
    {
        ReadyPlayers = bytes[offset];
    }

    protected override void WritePayload(byte[] bytes, byte offset)
    {
        bytes[offset] = ReadyPlayers;
    }

    public override string ToString()
    {
        return $"ReadyPlayerCount{{ readyPlayerCount={ReadyPlayers} }}";
    } 
}

